### PR TITLE
AICH-2878 [v8][Comquest] Allow Bot Launch with Default Intent

### PR DESCRIPTION
**Pull request by bhupesh-lad-cognoai**

### CODE REVIEW

File: CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift [hunk 1]
```diff
@@ -254,6 +254,32 @@ public class ChatbotSDK: UIViewController, UIWebViewDelegate, WKUIDelegate, WKNa
         return nil
     }
 
+    private func chatbotIndexURL(includeCustomParams: Bool = false) -> URL? {
```
Comment (bhupesh-lad-cognoai) [inline CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift]: **[Test Coverage]** **What's happening**: The PR adds a new URL-building helper that decides which query parameters are sent for initial load versus reload flows.

**What's wrong**: No test files were changed, and GitHub code search did not find an existing XCTest target. This URL behavior is the core of the feature, so regressions in query parameter inclusion could ship unnoticed.

Please add focused tests or documented manual verification covering initial launch, reload, live-chat reload, empty `customParams`, and `customParams` with/without a leading `&`.

> **Behavioral change**: No — adding tests or explicit manual verification does not change runtime behavior, but it protects the URL contract introduced by this PR.

File: CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift [hunk 2]
```diff
@@ -321,15 +347,15 @@ extension ChatbotSDK: WKScriptMessageHandler {
             // comment this line as user id should not be reset while reloading chatbot
             // Constants.mobileChatbotSessionID  = ""
             Constants.mobileLiveChatSessionID = ""
-            if let _url = URL(string: Constants.botUrl + "/chat/index/?id=" + Constants.botId + "&channel=iOS&mobile_session_id=" + Constants.mobileChatbotSessionID + "&livechat_session_id=" + Constants.mobileLiveChatSessionID + "&selected_language=" + Constants.chatbotSelectedLanguage) {
+            if let _url = chatbotIndexURL() {
```
Comment (bhupesh-lad-cognoai) [inline CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift]: **[Logic & Correctness]** **What's happening**: The new helper always appends `campaign_link_query_id` and `hide_welcome_message_at_start` whenever those constants are set. The reload paths now call this helper too.

**What's wrong**: These look like launch-time/default-intent parameters. Passing them again during `reloadChatbot` or `reloadChatbotForLiveChat` can relaunch the default intent or suppress welcome behavior during reload flows, not just on the initial bot launch.

A safer shape is to scope those params explicitly to the initial load, for example by calling `chatbotIndexURL(includeCustomParams: true, includeLaunchParams: true)` from the initial webview load, adding an `includeLaunchParams` parameter to `chatbotIndexURL`, and guarding the two new launch-only query params behind that flag.

> **Behavioral change**: Yes — default-intent launch parameters are sent only on the first webview load, while reload/live-chat reload flows keep their previous behavior.

Comment (bhupesh-lad-cognoai) [reply CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift]: This was discussed with product this is expecet behavior

File: CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift (hunk 1 above)
Comment (bhupesh-lad-cognoai) [reply CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift]: Current SDK dont have any test suite added, will have to implement from scratch. should be taken separately

File: CognoChatbot-iOS/Exo-CognoChatbot-iOS.podspec [hunk 3]
```diff
@@ -1,7 +1,7 @@
 Pod::Spec.new do |spec|
 
   spec.name         = "Exo-CognoChatbot-iOS"
-  spec.version      = "1.0.29"
+  spec.version      = "1.0.30"
```
Comment (kusal-saraf-cognoai) [inline CognoChatbot-iOS/Exo-CognoChatbot-iOS.podspec]: @bhupesh-lad-cognoai Can we change the version to 8.0.1? Since this is specifically for v8 and v7 has a separate package

File: CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift [hunk 4]
```diff
@@ -321,15 +347,15 @@ extension ChatbotSDK: WKScriptMessageHandler {
...
                 let request = URLRequest(url: _url)
                 webViewGlobal.load(request)
             }
         } else if message.name == "reloadChatbotForLiveChat" {
             
             DispatchQueue.main.asyncAfter(deadline: .now() + 0.1){
                 
-                if let _url = URL(string: Constants.botUrl + "/chat/index/?id=" + Constants.botId + "&channel=iOS&mobile_session_id=" + Constants.mobileChatbotSessionID + "&livechat_session_id=" + Constants.mobileLiveChatSessionID + "&selected_language=" + Constants.chatbotSelectedLanguage) {
```
Comment (kusal-saraf-cognoai) [inline CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift:332]: @bhupesh-lad-cognoai Can we centralize the URL construction so that Constants.customParams is always appended? This will help ensure it doesn't get missed in future URL additions or modifications.

File: CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift [hunk 5]
```diff
@@ -321,15 +347,15 @@ extension ChatbotSDK: WKScriptMessageHandler {
             // comment this line as user id should not be reset while reloading chatbot
             // Constants.mobileChatbotSessionID  = ""
             Constants.mobileLiveChatSessionID = ""
-            if let _url = URL(string: Constants.botUrl + "/chat/index/?id=" + Constants.botId + "&channel=iOS&mobile_session_id=" + Constants.mobileChatbotSessionID + "&livechat_session_id=" + Constants.mobileLiveChatSessionID + "&selected_language=" + Constants.chatbotSelectedLanguage) {
```
Comment (kusal-saraf-cognoai) [inline CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift:324]: @bhupesh-lad-cognoai same here also

**Review by kusal-saraf-cognoai — CHANGES_REQUESTED** [review]

File: CognoChatbot-iOS/Exo-CognoChatbot-iOS.podspec (hunk 3 above)
Comment (bhupesh-lad-cognoai) [reply CognoChatbot-iOS/Exo-CognoChatbot-iOS.podspec]: Updated the podspec version to 8.0.1 in commit 8f2cdec.

File: CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift (hunk 4 above)
Comment (bhupesh-lad-cognoai) [reply CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift:332]: Updated this reload path to use chatbotIndexURL(includeCustomParams: true), so Constants.customParams is appended through the centralized URL helper in commit 8f2cdec.

File: CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift (hunk 5 above)
Comment (bhupesh-lad-cognoai) [reply CognoChatbot-iOS/CognoChatbot-iOS/ChatbotSDK.swift:324]: Updated this path as well to use chatbotIndexURL(includeCustomParams: true) in commit 8f2cdec.

**Review by kusal-saraf-cognoai — APPROVED** [review]
